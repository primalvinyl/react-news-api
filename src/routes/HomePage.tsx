import React from 'react';
import MasterTemplate from '../components/_MasterTemplate';
import Header from '../components/Header';
import SearchResults from '../components/SearchResults';
import styles from './HomePage.module.scss';

const HomePage = (): React.ReactElement => {
    return (
        <MasterTemplate>
            <div className={styles.root}>
                <Header />
                <SearchResults />
            </div>
        </MasterTemplate>
    );
};

export default HomePage;
