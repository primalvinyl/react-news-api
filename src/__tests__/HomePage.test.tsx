import React from 'react';
import { shallow } from 'enzyme';
import Renderer from 'react-test-renderer';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import rootReducer from '../store/reducers';
import HomePage from '../routes/HomePage';

describe('HomePage', () => {
    const store = createStore(rootReducer, {});

    it('renders', () => {
        const myWrapper = shallow(
            <Provider store={store}>
                <MemoryRouter>
                    <HomePage />
                </MemoryRouter>
            </Provider>
        );
        expect(myWrapper.exists());
    });
    it('matches snapshot', () => {
        const component = Renderer.create(
            <Provider store={store}>
                <MemoryRouter>
                    <HomePage />
                </MemoryRouter>
            </Provider>
        );
        expect(component.toJSON()).toMatchSnapshot();
    });
});
