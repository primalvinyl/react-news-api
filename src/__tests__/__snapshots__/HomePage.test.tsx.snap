// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage matches snapshot 1`] = `
<div
  className="root"
>
  <div
    className="root"
  >
    <header
      className="root"
    >
      <div
        className="wrapper"
      >
        <section
          className="root"
        >
          <form
            action="#"
            onReset={[Function]}
            onSubmit={[Function]}
          >
            <div
              className="root"
            >
              <div
                className="formControlWrapper"
              >
                <input
                  disabled={false}
                  id="searchField"
                  onBlur={[Function]}
                  onChange={[Function]}
                  required={false}
                  type="search"
                  value=""
                />
              </div>
            </div>
            <div
              className="root"
            >
              <div
                className="formControlWrapper"
              >
                <select
                  disabled={false}
                  id="searchSort"
                  onBlur={[Function]}
                  onChange={[Function]}
                  required={false}
                  value="date"
                >
                  <option
                    value="date"
                  >
                    Sort by Date
                  </option>
                  <option
                    value="relevance"
                  >
                    Sort by Relevance
                  </option>
                  <option
                    value="popularity"
                  >
                    Sort by Popularity
                  </option>
                </select>
              </div>
            </div>
            <div
              className="root"
            >
              <button
                disabled={false}
                id="searchButton"
                onClick={[Function]}
                style={
                  Object {
                    "background": "errorColor",
                  }
                }
                type="submit"
              >
                Search
              </button>
            </div>
          </form>
        </section>
      </div>
    </header>
    <main
      className="root"
    >
      <div
        className="wrapper"
      />
    </main>
  </div>
</div>
`;
